packages/@aws-cdk/aws-s3-deployment/lambda/src/index.py

```diff
@@ -120,12 +120,13 @@ def s3_deploy(s3_source_zip, s3_dest):
     os.mkdir(contents_dir)
 
     # download the archive from the source and extract to "contents"
-    archive=os.path.join(workdir, 'archive.zip')
-    logger.info("| archive: %s" % archive)
-    aws_command("s3", "cp", s3_source_zip, archive)
-    logger.info("| extracting archive to: %s" % contents_dir)
-    with ZipFile(archive, "r") as zip:
-      zip.extractall(contents_dir)
+    for s3_source_zip in s3_source_zips:
+        archive=os.path.join(workdir, str(uuid4()))
+        logger.info("archive: %s" % archive)
+        aws_command("s3", "cp", s3_source_zip, archive)
+        logger.info("| extracting archive to: %s\n" % contents_dir)
```

Comment:
I am trying to understand if it makes senes to use a different file name for each file instead of overwriting the same file...

Reply:
shit. just realized you already used a different file per source. I think that should be the behavior. the test expectation confused me... apologies, can you revert?